users/views.py

```diff
@@ -70,46 +71,144 @@ def user_logout(request):
+            form.save()
+            return redirect('home')
+
+    return render(request, "update_profile/update_form.html", {'form': form})
+
+
+@login_required(login_url="login")
+def main_page(request, date=None):
+
+    if not date:
+        year, month = None, None
+    else:
+        year, month = date.split("|")
+        year = int(year)
+        month = int(month)
+
+    cal = get_calendar_days(year, month)
+    current_month = get_current_month_name(month)
+    current_date = (datetime.now().year, datetime.now().month, datetime.now().day)
+    current_events = Event.objects.filter(eventparticipant__user_id=request.user.id)
+    table_measurements = get_table_measurements(current_events)
     week_days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
+
     return render(
         request, "user_site/home.html",
-        {'week_days': week_days, 'calandar': cal, 'month_name': current_month, 'current_date': current_date}
+        {'week_days': week_days, 'calendar': cal, 'month_name': current_month,
+         'next_date': get_dates(year, month, True), 'previous_date': get_dates(year, month, False),
+         'max_height': table_measurements[0], 'max_margin': table_measurements[1],
+         'max_margin_events': table_measurements[2], 'max_padding': table_measurements[3],
+         'current_date': current_date, 'current_events': current_events}
     )
 
 
+def get_dates(year=None, month=None, next=True):
+    """ return the next or previous date.
+        next: if true the next date will be returned otherwise the previous
+    """
+
+    if not year:
+        year = int(datetime.now().year)
+    if not month:
+        month = int(datetime.now().month)
+
+    if next:
+        action = 1
+    else:
+        action = -1
+
+    if month + action > 12:
+        year += 1
+        month -= 11
+    elif month + action < 1:
+        year -= 1
+        month += 11
+    else:
+        month += action
+
+    return f"{year}|{month}"
+
+
+def get_max_day_events(events):
+    """ return the maximum events count in one day over all the events given """
+
+    events_count = {}
+    if len(events) == 0:
+        return 0
+
+    for event in events:
+        if not events_count.get(event.date_time_start.date()):
+            events_count[event.date_time_start.date()] = 1
+        else:
+            events_count[event.date_time_start.date()] += 1
+
+    return max(events_count.values())
+
+
+def get_table_measurements(events):
+    """ get each tab in the table a measurement for the positioning
+
+        maximum row height, tab_top_positioning, event_margin_top, event_maring_bottom
+    """
+
+    max_amount_of_events = get_max_day_events(events)  # get max events in one day
+
+    if max_amount_of_events > 2:
+        tab_top_positioning = max_amount_of_events * 2
+        maximum_row_height = max_amount_of_events * 5
+    else:
+        maximum_row_height = 13
+        tab_top_positioning = 4
```

Comment:
maybe you can do it in 2 lines like:
```suggestion
    tab_top_positioning = max_amount_of_events * 2 if max_amount_of_events > 2 else 4
    maximum_row_height = max_amount_of_events * 5 if max_amount_of_events > 2 else 13
```

Reply:
done. thanks 😄